File: mobile/with-flutter/lib/config/alchemy_config.dart
import 'package:flutter_dotenv/flutter_dotenv.dart';

/// Configuration for the Account Abstraction (Alchemy) demo.
///
/// Credentials live in the `.env` file. See `.env.example` for required keys.
class AlchemyConfig {
  /// Alchemy API key used to talk to Alchemy's AA infrastructure.
  static String get apiKey => dotenv.env['ALCHEMY_API_KEY'] ?? '';

  /// Gas Manager policy ID used to sponsor user operations.
  static String get gasPolicyId => dotenv.env['ALCHEMY_GAS_POLICY_ID'] ?? '';

  /// Ethereum Sepolia testnet chain id.
  static const int sepoliaChainId = 11155111;

  /// Arbitrary burn address used as the destination for the demo transaction.
  static const String burnAddress = '0x000000000000000000000000000000000000dEaD';

  /// Sepolia block explorer base URL — used to link to a sent transaction.
  static const String sepoliaExplorerBaseUrl = 'https://sepolia.etherscan.io';

  static bool get isConfigured => apiKey.isNotEmpty;
}

File: mobile/with-swift/example/SmartAccount/AlchemyConfig.swift
import Foundation

/// Configuration for the Account Abstraction demo (Alchemy on Sepolia).
///
/// These values are sandbox-only credentials sourced from the external wallet example
/// (`~/para/web-sdk/examples/external-wallet-example/.env.sandbox`). Replace them with
/// your own Alchemy app and gas policy when running against beta or production.
enum AlchemyConfig {
    // Sandbox-only Alchemy credentials — replace for production.
    static let apiKey = "sfcC3PFq6Ma9d6jInNMRs"
    static let gasPolicyId = "06303734-b990-4510-a474-0ab5df919f74"

    /// Sepolia.
    static let chainId = 11_155_111

    /// Burn address used to demonstrate a sponsored zero-value transfer.
    static let burnAddress = "0x000000000000000000000000000000000000dEaD"

    /// Base URL for Sepolia Etherscan, used to link out to mined transactions.
    static let explorerBaseURL = "https://sepolia.etherscan.io"
}
